WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/bloggingprompts/BloggingPromptCardSource.kt

```diff
@@ -0,0 +1,107 @@
+package org.wordpress.android.ui.mysite.cards.dashboard.bloggingprompts
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MediatorLiveData
+import androidx.lifecycle.MutableLiveData
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.store.bloggingprompts.BloggingPromptsStore
+import org.wordpress.android.modules.BG_THREAD
+import org.wordpress.android.ui.mysite.MySiteSource.MySiteRefreshSource
+import org.wordpress.android.ui.mysite.MySiteUiState.PartialState.BloggingPromptUpdate
+import org.wordpress.android.ui.mysite.SelectedSiteRepository
+import org.wordpress.android.util.config.BloggingPromptsFeatureConfig
+import java.util.Date
+import javax.inject.Inject
+import javax.inject.Named
+
+const val REFRESH_DELAY = 500L
+
+class BloggingPromptCardSource @Inject constructor(
+    private val selectedSiteRepository: SelectedSiteRepository,
+    private val promptsStore: BloggingPromptsStore,
+    private val bloggingPromptsFeatureConfig: BloggingPromptsFeatureConfig,
+    @param:Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher
+) : MySiteRefreshSource<BloggingPromptUpdate> {
+    override val refresh = MutableLiveData(false)
+
+    companion object {
+        private const val NUM_PROMPTS_TO_REQUEST = 20
+    }
+
+    override fun build(coroutineScope: CoroutineScope, siteLocalId: Int): LiveData<BloggingPromptUpdate> {
+        val result = MediatorLiveData<BloggingPromptUpdate>()
+        result.getData(coroutineScope, siteLocalId)
+        result.addSource(refresh) { result.refreshData(coroutineScope, siteLocalId, refresh.value) }
+        refresh()
+        return result
+    }
+
+    private fun MediatorLiveData<BloggingPromptUpdate>.getData(
+        coroutineScope: CoroutineScope,
+        siteLocalId: Int
+    ) {
+        val selectedSite = selectedSiteRepository.getSelectedSite()
+        if (selectedSite != null && selectedSite.id == siteLocalId && bloggingPromptsFeatureConfig.isEnabled()) {
+            coroutineScope.launch(bgDispatcher) {
+                promptsStore.getPromptForDate(selectedSite, Date()).collect { result ->
+                    postValue(BloggingPromptUpdate(result.model))
+                }
+            }
+        } else {
+            postErrorState()
+        }
+    }
+
+    private fun MediatorLiveData<BloggingPromptUpdate>.refreshData(
+        coroutineScope: CoroutineScope,
+        siteLocalId: Int,
+        isRefresh: Boolean? = null
```

Comment:
💡 Since the same happens if `isRefresh` value is `null` or `true`, do you think it would make sense to make this parameter not nullable and with `true` as default?
```
...
isRefresh: Boolean = true
```

Reply:
I copied this logic from `CardSource`, and looks like it's done like this because we pass the value from LiveDate which is nullable, so we either have to do this `null, true -> refreshData` or do something with the value before passing it to the method :)